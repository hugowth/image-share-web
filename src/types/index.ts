export type LoginPayload = {
  username: string;
  password: string;
};

export type User = {
  id: string;
  username: string;
};

export type Post = {
  id: string;
  created_by: User;
  create_date: string;
  description: string;
  image: string;
  update_date: string;
};

export type PostPayload = {
  description: string;
  image: File | undefined;
};
