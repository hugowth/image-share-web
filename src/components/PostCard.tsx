import React from "react";
import {
  Card,
  CardHeader,
  CardMedia,
  CardContent,
  Typography,
  Avatar,
} from "@material-ui/core";
import { red } from "@material-ui/core/colors";
import { makeStyles, Theme } from "@material-ui/core/styles";
import { Post } from "../types";

const useStyles = makeStyles((theme: Theme) => ({
  root: {
    maxWidth: 345,
    margin: 12,
  },
  media: {
    height: 0,
    paddingTop: "56.25%", // 16:9
  },
  expand: {
    transform: "rotate(0deg)",
    marginLeft: "auto",
    transition: theme.transitions.create("transform", {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: "rotate(180deg)",
  },
  avatar: {
    backgroundColor: red[500],
  },
}));

type Props = {
  post: Post;
};

const PostCard = ({ post }: Props) => {
  const classes = useStyles();

  return (
    <Card className={classes.root}>
      <CardHeader
        avatar={<Avatar aria-label="recipe">R</Avatar>}
        title={post.description}
        subheader={post.create_date}
      />
      {post.image && <CardMedia className={classes.media} image={post.image} />}
      <CardContent>
        <Typography color="textSecondary" gutterBottom>
          created_by: {post.created_by.username}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default PostCard;
