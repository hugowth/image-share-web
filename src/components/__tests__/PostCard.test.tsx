import React from "react";
import renderer from "react-test-renderer";

import PostCard from "../PostCard";
import { Post, User } from "../../types";

describe("PostCard", () => {
  const user: User = {
    id: "1",
    username: "test",
  };
  const post: Post = {
    id: "1",
    created_by: user,
    create_date: "2021-07-10T06:08:21.506566Z",
    description: "test",
    image:
      "http://p37-env.eba-vvnpnaez.ap-southeast-1.elasticbeanstalk.com/upload/upload/signal-2021-03-16-225319_002.jpeg",
    update_date: "2021-07-10T06:08:21.506566Z",
  };
  it("renders correctly", () => {
    const tree = renderer.create(<PostCard post={post} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
