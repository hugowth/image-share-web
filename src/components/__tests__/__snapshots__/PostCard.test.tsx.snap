// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-avatar"
    >
      <div
        aria-label="recipe"
        className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
      >
        R
      </div>
    </div>
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
      >
        test
      </span>
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        2021-07-10T06:08:21.506566Z
      </span>
    </div>
  </div>
  <div
    className="MuiCardMedia-root makeStyles-media-2"
    style={
      Object {
        "backgroundImage": "url(\\"http://p37-env.eba-vvnpnaez.ap-southeast-1.elasticbeanstalk.com/upload/upload/signal-2021-03-16-225319_002.jpeg\\")",
      }
    }
  />
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
    >
      created_by: 
      test
    </p>
  </div>
</div>
`;
